addon/lib/connect.js

```diff
@@ -0,0 +1,46 @@
+import Ember from 'ember';
+import reduxStore from '../services/redux-store';
+
+const {
+  computed,
+  defineProperty,
+  inject: { service }
+} = Ember;
+
+export default function connect(mapStateToComputed, mapDispatchToActions, storeService) {
+  return function wrapWithConnect(WrappedComponent) {
+    return WrappedComponent.extend({
+      reduxStore: storeService || service('reduxStore'),
+
+      init() {
+        this._super(...arguments);
```

Comment:
How did you decide where to put the call to `this._super` (i.e., top or bottom of function) and whether or not to `return` it?